File: treatment/src/main/java/com/hartwig/actin/treatment/trial/config/InclusionCriteriaConfigFactory.java
package com.hartwig.actin.treatment.trial.config;

import java.util.Map;

import com.hartwig.actin.treatment.trial.TrialConfigDatabaseUtil;

import org.jetbrains.annotations.NotNull;

public class InclusionCriteriaConfigFactory implements TrialConfigFactory<InclusionCriteriaConfig> {

    @NotNull
    @Override
    public InclusionCriteriaConfig create(@NotNull Map<String, Integer> fields, @NotNull String[] parts) {
        return ImmutableInclusionCriteriaConfig.builder()
                .trialId(parts[fields.get("trialId")])
                .criterionIds(TrialConfigDatabaseUtil.toCriterionIds(parts[fields.get("criterionIds")]))
                .appliesToCohorts(TrialConfigDatabaseUtil.toCohorts(parts[fields.get("appliesToCohorts")]))
                .inclusionRule(parts[fields.get("inclusionRule")])
                .build();
    }
}

File: treatment/src/main/java/com/hartwig/actin/treatment/trial/config/InclusionCriteriaConfigFactory.kt
package com.hartwig.actin.treatment.trial.config

import com.hartwig.actin.treatment.trial.TrialConfigDatabaseUtil

class InclusionCriteriaConfigFactory : TrialConfigFactory<InclusionCriteriaConfig> {
    override fun create(fields: Map<String, Int>, parts: Array<String>): InclusionCriteriaConfig {
        return InclusionCriteriaConfig(
            trialId = parts[fields["trialId"]!!],
            referenceIds = TrialConfigDatabaseUtil.toReferenceIds(parts[fields["referenceIds"]!!]),
            appliesToCohorts = TrialConfigDatabaseUtil.toCohorts(parts[fields["appliesToCohorts"]!!]),
            inclusionRule = parts[fields["inclusionRule"]!!]
        )
    }
}